src/components/vr_cooperation/src/register_service_request.cc

```diff
@@ -60,7 +60,8 @@ void RegisterServiceRequest::Execute() {
   int result_code;
   bool success = true;
   std::string info;
-  if (parent_->IsVRServiceSupported()) {
+  VRModule* parent = GetVRModule();
+  if (parent->IsVRServiceSupported()) {
```

Comment:
I believe that using of this condition should be in place where all messages from mobile are handled. Because in your way you will have double code in different classes of commands.